packages/jellyfish-api-core/__tests__/category/loan/listVaultHistory.test.ts

```diff
@@ -0,0 +1,778 @@
+      expect(vaultHistory[3].txid).toStrictEqual(depositVault2TxId)
+      expect(vaultHistory[3].blockHeight).toStrictEqual(depositVault2BlockHeight)
+      expect(vaultHistory[3].amounts).toStrictEqual(['-0.50000000@BTC']) // Deposit 0.5 BTC
+
+      // 4 - Take loan
+      expect(vaultHistory[4].type).toStrictEqual('TakeLoan')
+      expect(vaultHistory[4].txid).toStrictEqual(takeLoanTxId)
+      expect(vaultHistory[4].blockHeight).toStrictEqual(takeLoanBlockHeight)
+      expect(vaultHistory[4].amounts).toStrictEqual(['7500.00000000@AAPL']) // TakeLoan 7500 AAPL
+
+      // 5 - Payback loan
+      expect(vaultHistory[5].type).toStrictEqual('PaybackLoan')
+      expect(vaultHistory[5].txid).toStrictEqual(paybackLoanTxId)
+      expect(vaultHistory[5].blockHeight).toStrictEqual(paybackLoanBlockHeight)
+      expect(vaultHistory[5].amounts).toStrictEqual(['0.00057078@DFI'])
+
+      // 6 - Payback loan
+      expect(vaultHistory[6].type).toStrictEqual('PaybackLoan')
+      expect(vaultHistory[6].txid).toStrictEqual(paybackLoanTxId)
+      expect(vaultHistory[6].blockHeight).toStrictEqual(paybackLoanBlockHeight)
+      expect(vaultHistory[6].amounts).toStrictEqual(['-500.00000000@AAPL'])
+
+      // 7 - Withdraw from vault
+      expect(vaultHistory[7].type).toStrictEqual('WithdrawFromVault')
+      expect(vaultHistory[7].address).toStrictEqual(aliceColAddr)
+      expect(vaultHistory[7].txid).toStrictEqual(withdrawFromVaultTxId)
+      expect(vaultHistory[7].blockHeight).toStrictEqual(withdrawFromVaultBlockHeight)
+      expect(vaultHistory[7].amounts).toStrictEqual(['50.00000000@DFI'])
+
+      // 8 - Liquidation
+      expect(vaultHistory[8].vaultSnapshot?.state).toStrictEqual('inLiquidation')
+      expect(vaultHistory[8].blockHeight).toStrictEqual(inLiquidationBlockHeight)
+      expect(vaultHistory[8].vaultSnapshot?.collateralAmounts).toStrictEqual([]) // No collateralAmounts for liquidation
+      expect(vaultHistory[8].vaultSnapshot?.collateralValue).toStrictEqual(0) // No collateralValue for liquidation
+      expect(vaultHistory[8].vaultSnapshot?.batches).toStrictEqual([
+        {
+          index: 0,
+          collaterals: [
+            '6655.51838400@DFI',
+            '0.33444816@BTC'
+          ],
+          loan: '4682.28594829@AAPL'
+        },
+        {
+          index: 1,
+          collaterals: [
+            '3294.48161600@DFI',
+            '0.16555184@BTC'
+          ],
+          loan: '2317.73155561@AAPL'
+        }
+      ])
+
+      // 9 - Auction bid
+      expect(vaultHistory[9].type).toStrictEqual('AuctionBid')
+      expect(vaultHistory[9].address).toStrictEqual(aliceColAddr)
+      expect(vaultHistory[9].txid).toStrictEqual(auctionBidTxId1)
+      expect(vaultHistory[9].blockHeight).toStrictEqual(auctionBidBlockHeight1)
+      expect(vaultHistory[9].amounts).toStrictEqual(['-5000.00000000@AAPL'])
+
+      if (vaultHistory[10].address === bobColAddr) {
```

Comment:
2 entries are made if a bid outbid by another bid.
However the order is inconsistent.